src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/DetailsVA.util.jsx

```diff
@@ -0,0 +1,25 @@
+import { APPOINTMENT_STATUS } from '../../../utils/constants';
+
+/**
+ * Returns appointment type header.
+ *
+ * @param {object} appointment appoinment object
+ * @returns {string} returns string
+ */
+export function formatHeader(appointment) {
+  const { isPastAppointment, isCompAndPenAppointment } = appointment.vaos;
+  const cancelled = appointment.status === APPOINTMENT_STATUS.cancelled;
+  if (appointment.vaos.isCOVIDVaccine) {
+    return 'COVID-19 vaccine';
+  }
+  if (appointment.vaos.isPhoneAppointment) {
+    return 'VA appointment over the phone';
+  }
+  if (
+    (isPastAppointment && isCompAndPenAppointment) ||
+    (isCompAndPenAppointment && cancelled)
+  ) {
```

Comment:
I think you can simply test for isCompAndPenAppointment. No need to check if it's also cancelled or past appointment.

Reply:
@jenniemc - the reason I included this check was because for upcoming appts, we do not display `TypeHeader`.  TypeHeader is the header above the facility address. Please see screenshots above.

Reply:
@simiadebowale, sounds like there are two checks; one in DetailsVA.util.jsx which returns the value "VA Appointment" for upcoming C&P as the TypeHeader. My suggested change returns "Compensation and pension exam" as TypeHeader. But neither matters because the main check is in DetailsVA.jsx using the boolean `displayTypeHeader` which does not show the TypeHeader if it's an upcoming C&P appointment.
```
  // we don't want to display the appointment type header for upcoming C&P appointments.
  const displayTypeHeader =
    !isCompAndPenAppointment ||
    (isCompAndPenAppointment && (isPastAppointment || canceled));
...
      {displayTypeHeader && <TypeHeader>{header}</TypeHeader>}
```
Your current logic is good and works as expected.